python/ray/data/datasource/file_based_datasource.py

```diff
@@ -468,92 +465,8 @@ def _add_partitions_to_dataframe(
     return df
 
 
-def _wrap_s3_serialization_workaround(filesystem: "pyarrow.fs.FileSystem"):
-    # This is needed because pa.fs.S3FileSystem assumes pa.fs is already
-    # imported before deserialization. See #17085.
-    import pyarrow as pa
-    import pyarrow.fs
-
-    wrap_retries = False
-    fs_to_be_wrapped = filesystem  # Only unwrap for S3FileSystemWrapper
-    retryable_errors = []
-    if isinstance(fs_to_be_wrapped, RetryingPyFileSystem):
-        wrap_retries = True
-        retryable_errors = fs_to_be_wrapped.retryable_errors
-        fs_to_be_wrapped = fs_to_be_wrapped.unwrap()
-    if isinstance(fs_to_be_wrapped, pa.fs.S3FileSystem):
-        return _S3FileSystemWrapper(
-            fs_to_be_wrapped,
-            wrap_retries=wrap_retries,
-            retryable_errors=retryable_errors,
-        )
-    return filesystem
-
-
-def _unwrap_s3_serialization_workaround(
-    filesystem: Union["pyarrow.fs.FileSystem", "_S3FileSystemWrapper"],
-):
-    if isinstance(filesystem, _S3FileSystemWrapper):
-        wrap_retries = filesystem._wrap_retries
-        retryable_errors = filesystem._retryable_erros
-        filesystem = filesystem.unwrap()
-        if wrap_retries:
-            filesystem = RetryingPyFileSystem.wrap(
-                filesystem, retryable_errors=retryable_errors
-            )
-    return filesystem
-
-
-class _S3FileSystemWrapper:
-    def __init__(
-        self,
-        fs: "pyarrow.fs.S3FileSystem",
-        wrap_retries: bool = False,
-        retryable_errors: List[str] = tuple(),
-    ):
-        self._fs = fs
-        self._wrap_retries = wrap_retries
-        self._retryable_erros = retryable_errors
-
-    def unwrap(self):
-        return self._fs
-
-    @classmethod
-    def _reconstruct(cls, fs_reconstruct, fs_args):
-        # Implicitly trigger S3 subsystem initialization by importing
-        # pyarrow.fs.
-        import pyarrow.fs  # noqa: F401
-
-        return cls(fs_reconstruct(*fs_args))
-
-    def __reduce__(self):
-        return _S3FileSystemWrapper._reconstruct, self._fs.__reduce__()
```

Comment:
the original bug I was going to fix was this: this thing doesn't keep the wrap_retries, retryable errors args properly

But I think none of this code is needed anymore. Is there a min version of pyarrow that Ray Data supports?

Reply:
the min version we currently support is arrow 9.